### DISCUSSION TITLE
Smart-Home Integration durch EET?

### DISCUSSION THREAD

**Discussion by axcro7** [mmattel/EET-Solmate#59] [General]
Ich vermute die meisten hier sind sowieso aus AT/DE, daher erspare ich euch meine Englischkünste ;)

Im Newsletter vom 4.7.2024 heißt es:
> Mit Ende des Monats bekommt SolMate ein Software-Update, das es möglich macht, SolMate mit Amazon Alexa bedienen. So kannst du SolMate zum Beispiel mit Befehlen wie “Alexa, aktiviere den Boost-Modus" steuern, sowie auch Informationen abrufen. **Außerdem arbeiten wir hart daran, SolMate generell Smart-Home-tauglich zu machen.**

Ist vielleicht jemand mit den Entwicklern bei EET in Kontakt? wäre spannend zu erfahren was die genau vor haben - eventuell würde das ja auch die Arbeit an der HA Integration erleichtern?

Grüße aus NÖ

**Comment by mmattel** [comment]
Ich hab "etwas" Kontakt zum SW Entwicklungsleiter von EET und kann die Ankündigung bestätigen obwohl ich da nichts genaues weis und keine Details habe. Zum Eigenlob, meine SW ist nach deren Rückmeldung bei mir da wohl die Messlatte... Ich bin hier sehr gespannt wie EET das umsetzen wird denn ich kann aus eigener Erfahrung sagen dass das nicht einfach ist und man recht viele Parameter und Eigentümlichkeiten der "Fruchtkette" beachten muss. HA ist neben EET nicht gerade up to date und auskunftsfreudig was Dokumentation anbelangt.

Grüße aus Wien 👋

**Comment by axcro7** [comment]
Okay, sowas in der Art hatte ich schon vermutet .. 
Also dann rate ich mal ins Blaue und vermute, sie werden (zunächst) nur eine Cloud-Anbindung über deren Server anbieten und weniger eine lokale Integration.

Danke für die vielen Stunden die du da bestimmt schon reingesteckt hast! Ich habe mit den Code mal um Ostern herum angeschaut und für mich als "Laie" (bin Sysadmin und kein Programmierer) schaut das durchaus nach einer hohen Messlatte aus :)

**Reply by mmattel** [reply]
@axcro7 Zur Info, eine neue Release v6 meiner SW ist da 😄

**Comment by mmattel** [comment]
@axcro7 Es gibt Neuigkeiten ! 😃 Siehe [v7.0.0](https://github.com/mmattel/EET-Solmate/releases/tag/v7.0.0)

**Comment by axcro7** [comment]
kleines Update meinerseits bezüglich der Smarthome "Integration" seitens EET: 

seit sun2plug 0.5.1 wird die direkte Verbindung mit einem Shelly 3EM, Shelly Pro 3EM oder Shelly Plus Plug S unterstützt. Der SolMate fragt dann regelmäßig die Shelly Geräte ab und richtet sich dann bei der Einspeisung nur noch danach.
Ich verwende das seither mit meinem Shelly 3EM und muss sagen, es funktioniert wirklich gut: Ich habe den Einspeisebereich wieder auf 0-800W eingestellt (kein Profil mehr) und wir brauchen den Boost Modus auch gar nicht mehr. Seither schenken wir deutlich weniger Strom her! (nur wenn der Akku voll ist und der Ertrag höher ist als der Verbrauch)

Was aber nirgendwo erwähnt wird: Wenn der Login am Shelly Gerät mit einem Passwort geschützt ist, funktioniert es einfach nicht. Kein Hinweis wie "Authentifizierungsproblem / falsches Passwort" oder sonst irgendwas hilfreiches ...

**Reply by mmattel** [reply]
Einige der esham Nutzer verwenden NodeRED um genau das zu machen und sind sehr erfolgreich damit. Vorallem läuft das innerhalb HA und da hast Du dann das "Gewürks" mit Zugriffsrechten nicht mehr und bist außerdem deutlich flexibler. Ich würde NIE auf die Idee kommen meine Sicherheitsstufe zu reduzieren nur um einem Gerät Zugriff zu gestatten - Never-Ever!!!

**Reply by axcro7** [reply]
Auch wenn es Spaß machen würde, ich habe aktuell leider nicht die Zeit mir da selbst etwas zu basteln.. daher kommt mir das sehr entgegen. Außerdem bin ich dadurch nicht von einem externen Dienst abhängig, sprich sollte der HA ausfallen, funktioniert es weiter.

Normalerweise geht bei mir auch nichts ohne PW und Verschlüsselung. In dem Fall ist wäre es aber keine wirkliche Verbesserung der Sicherheit das PW zu setzen, da mein 3EM ohnehin nur HTTP spricht. Meine IoT Geräte sind außerdem in einem abgetrennten VLAN und auf einer eigenen WLAN SSID mit Client Isolation unterwegs. Und auf der Firewall ist auch nur das notwendigste erlaubt.
Also für mich ein kalkuliertes Risiko.

**Reply by josy1024** [reply]
ich überlege mir auch eine bridge von mqtt auf eine "fake" shelly 3EM zu bauen.
hast du infos, welche api/routes solmate verwendet? @mmattel 
(winterprojekt für den nächsten frühling) ;-)

**Reply by mmattel** [reply]
Kannst Du beschreiben was Du erreichen möchtest? Mir ist das nicht klar.

**Reply by josy1024** [reply]
Ich vermute die Regelung ist über Leistungsmessgerät ist genauer als mit der "NetDetection". Von der EVN bekomm ich ja alle 6 Sekunden aktuelle Verbrauchstwerte über MQTT. 
Meine Idee ist, dass ich eine 3EM API emuliere und mit den MQTT Daten füttere. (Also MQTT to "virtual" shelly API)
Die Solmate greift dann die "virtual shelly" via API direkt ab. 
Damit könnte die Solmate genauer/schneller die Einspeiseleistung Regeln. 
Ich habe leider keine Mess-Shelly, damit ich mir das genauer anschauen kann. 
Nachdem EET ja auch auf eine Authentifierung zur Shelly verzichtet sollte das relativ einfach gehen.

Vielleicht bleibt es aber auch nur ein Ideenfragment und wird wieder verworfen, weil EET auch eine direkte MQTT Mess-Schnittstelle baut. mal schauen ...

**Reply by mmattel** [reply]
Der Solmate benötigt für Shelly kein MQTT. Der fährt ja direkt drauf. Du müsstest also ein komplettes reverse Engineering sowohl vom Solmate und speziell auch vom Shelly machen und das halte ich nicht für zielführend. Nicht dass ich Dich davon abhalten will... M.e. ist es wirklich besser und vor allem sehr viel einfacher die Einspeisewerte vom EV in MQTT einzuspeisen und dann zB mit NodeRED den Ablauf zu definieren. Just my 2c 😃

> weil EET auch eine direkte MQTT Mess-Schnittstelle baut.

davon hab ich noch nichts gehört, lass mich aber gerne überraschen. (Ich sag nur: Solmate tausch, Multi-Solmates etc.) Was mal kommt ist eine überarbeitete API...

**Reply by josy1024** [reply]
hallo @mmattel,

etwas später - hab die Integration heute mal runtergeschreiben.... (einfacher http request mit json output: http://xx.xx.xx.xx/status https://shelly-api-docs.shelly.cloud/gen1/#shelly-3em-status)
Verwendet flask + paho-mqtt

die solemate holt sich alle 5 sekunden über meinen raspi zero die aktuellen Leistungswerte der "Virtual Shelly" 

# Architektur
also Solmate -> Raspi  zero mit EET-Solmate + **(virtualshelly mit mqtt-client)** -> (HASS) Mqtt-server -> EVN [Sagemcom-T210-D] Mbus Zähler

# Solemate Einrichtung:
<img width="1233" height="763" alt="image" src="https://github.com/user-attachments/assets/6d6adc71-2f5a-4cf0-b82d-9c0bdba6d02e" />

https://github.com/josy1024/SmartMeterEVNSagemcom-T210-D/blob/main/virtualshelly.MD

Ich werde das ganze in den nächsten Tagen beobachten, wie gut Solmate jetzt den Energieverbrauch regelt ;-)
